dsp-shared/src/main/scala/dsp/valueobjects/Id.scala

```diff
@@ -110,3 +114,66 @@ object Id {
     }
   }
 }
+
+/**
+ * Stores the project ID, i.e. UUID, IRI and short code of the project
+ *
+ * @param uuid      the UUID of the project
+ * @param iri       the IRI of the project
+ * @param shortCode the shortcode of the project
+ */
+abstract case class ProjectId private (
+  uuid: UUID,
+  iri: Iri.ProjectIri,
+  shortCode: Project.ShortCode
+)
+
+/**
+ * Companion object for ProjectId. Contains factory methods for creating ProjectId instances.
+ */
+object ProjectId {
+
+  /**
+   * Generates a ProjectId instance with a new (random) UUID and an IRI which is created from a prefix and the UUID.
+   *
+   * @param iri       the project IRI
+   * @param shortCode the project short code (as defined by the ARK resolver)
+   * @return a new ProjectId instance
+   */
+  def fromIri(iri: Iri.ProjectIri, shortCode: Project.ShortCode): Validation[ValidationException, ProjectId] = {
+    val uuid: Try[UUID] = Try(UUID.fromString(iri.value.split("/").last))
+    val projectId: Either[ValidationException, ProjectId] = uuid.toEither.fold(
+      _ => Left(new ValidationException(IdErrorMessages.IriDoesNotContainUuid(iri))),
```

Comment:
Could you add a test to cover this case?

Reply:
this one I can test, you're right